### PR TITLE
Add workflows and fix tests

### PR DESCRIPTION
**Pull request by kasunidilshani**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                'https://api.github.com/repos/ballerina-platform/ballerina-release/dispatches' \
+                --header 'Accept: application/vnd.github.v3+json' \
+                --header 'Authorization: Bearer ${{ secrets.BALLERINA_BOT_TOKEN }}' \
+                --data-raw '{
+                  "event_type": "notify-ballerinax-connector-build-failure",
+                  "client_payload": {
+                    "repoName": "module-ballerinax-scim",
+                    "workflow": "CI"
+                  }
+                }'
```
Comment (Shaaali) [inline .github/workflows/ci.yml:60]: Add newline

Comment (Shaaali) [reply .github/workflows/ci.yml:60]: Check other places as well.

**Review by LakshanSS — COMMENTED** [review]
Checked the workflows. In both dev-stg release.yml and releas.yml, the ballerina version is addded as `2201.2.1`

This is an old version. As choreo supports `2201.4.1`, we can also release this connector with this version.
 We might have to fix this version in Ballerina.toml. 
 
 Other than this, all looks good to me.

File: .github/workflows/dev-stg-release.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+jobs:
+    release:
+
+        runs-on: ubuntu-latest
+        env: 
+          BALLERINA_${{ github.event.inputs.bal_central_environment }}_CENTRAL: true
+        steps:
+            - uses: actions/checkout@v2
+            - name: Ballerina Build
+              uses: ballerina-platform/ballerina-action/@2201.2.1
```
Comment (RolandHewage) [inline .github/workflows/dev-stg-release.yml:23]: Let's update this version with the specific Ballerina distribution u r planning to release from.
Define the specific version in the Ballerina.toml. Please apply the same in all the relevant places in the release.yml files
https://github.com/ballerina-platform/module-ballerinax-scim/blob/90c4d7d65893eb2e11a424cb2ca14a8f3ae9a600/scim/Ballerina.toml#L5

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            - name: Ballerina Tests
+              uses: ballerina-platform/ballerina-action@nightly
+              with:
+                  args:
+                      test --code-coverage ./scim
+              env:
+                  ORG_NAME: ${{ secrets.ORG_NAME }}
+                  CLIENT_ID: ${{ secrets.CLIENT_ID }}
+                  CLIENT_SECRET: ${{ secrets.CLIENT_SECRET }}
+
```
Comment (lnash94) [inline .github/workflows/ci.yml:30]: Shall we remove this additional new line?

**Review by LakshanSS — APPROVED** [review]

**Review by lnash94 — APPROVED** [review]
Checked the workflows, LGTM

**Pull request merged**